### PR TITLE
[Core] Upgrade azure-mgmt-core to 1.2.1

### PR DESCRIPTION
**Pull request by jsntcy**
**Description**  
Upgrade azure-mgmt-core to 1.2.1 due to codegen's upgrade, see details in #15766 

**Testing Guide**  

**History Notes**  

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.  
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Upgrade azure-mgmt-core

File: src/azure-cli/requirements.py3.Darwin.txt [hunk 1]
```diff
@@ -31,8 +31,8 @@ azure-mgmt-consumption==2.0.0
 azure-mgmt-containerinstance==1.5.0
 azure-mgmt-containerregistry==3.0.0rc15
 azure-mgmt-containerservice==9.4.0
-azure-mgmt-core==1.2.0
-azure-core==1.7.0
+azure-mgmt-core==1.2.1
+azure-core==1.8.2
```
Comment (jiasli) [inline src/azure-cli/requirements.py3.Darwin.txt:35]: `azure-mgmt-core` already specifies the dependency on `azure-core`.

https://github.com/Azure/azure-sdk-for-python/blob/e1001d7f7d9708b33f087c7a5ac652d1adeca89e/sdk/core/azure-mgmt-core/setup.py#L70

```py
    install_requires=[
        "azure-core<2.0.0,>=1.9.0",
    ],
```

Does CLI need to specify it directly?

Comment (jsntcy) [reply src/azure-cli/requirements.py3.Darwin.txt:35]: Besides azure-mgmt-core, data plane commands use azure.core directly, so we need specify the version for it.

Comment (jiasli) [reply src/azure-cli/requirements.py3.Darwin.txt:35]: Installing `azure-mgmt-core` will also install `azure-core`, and `1.8.2` doesn't fit the range `"azure-core<2.0.0,>=1.9.0"`.

Comment (jiasli) [reply src/azure-cli/requirements.py3.Darwin.txt:35]: That makes sense. 

`azure-mgmt-core` 1.2.1 has dependency

https://github.com/Azure/azure-sdk-for-python/blob/azure-mgmt-core_1.2.1/sdk/core/azure-mgmt-core/setup.py#L70
```py
"azure-core<2.0.0,>=1.8.2",
```

Make sure this dependency is always satisfied if we bump any libs.

**Review by jiasli — APPROVED** [review]

**Pull request merged**